src/config/findHomeDir.ts

```diff
@@ -0,0 +1,76 @@
+import fs from 'fs';
+import path from 'path';
+
+// Implement the same logic as in
+// https://github.com/kubernetes/kubernetes/blob/ea07644/staging/ \
+// src/k8s.io/client-go/util/homedir/homedir.go
+
+function dropDuplicatesAndNils(a: Array<string | null | undefined>): string[] {
+  return a.reduce((acceptedValues, currentValue) => {
+    // Good-enough algorithm for reducing a small (3 items at this point) array into an ordered list
+    // of unique non-empty strings.
+    if (currentValue && !acceptedValues.includes(currentValue)) {
+      return acceptedValues.concat(currentValue);
+    } else {
+      return acceptedValues;
+    }
+  }, [] as string[]);
+}
+
+export function findHomeDir(): string | null {
+  if (process.platform !== 'win32') {
+    if (process.env.HOME) {
+      try {
+        fs.accessSync(process.env.HOME);
+
+        return process.env.HOME;
+      } catch (ignore) { }
+    }
+
+    return null;
+  }
+
+  // $HOME is always favored, but the k8s go-client prefers the other two env vars
+  // differently depending on whether .kube/config exists or not.
+  const homeDrivePath = process.env.HOMEDRIVE && process.env.HOMEPATH ? path.join(process.env.HOMEDRIVE, process.env.HOMEPATH) : null;
+  const favourHomeDrivePathList: string[] =
+    dropDuplicatesAndNils([process.env.HOME, homeDrivePath, process.env.USERPROFILE]);
+  const favourUserProfileList: string[] =
+    dropDuplicatesAndNils([process.env.HOME, process.env.USERPROFILE, homeDrivePath]);
+
+  // 1. the first of %HOME%, %HOMEDRIVE%%HOMEPATH%, %USERPROFILE% containing a `.kube/config` file is returned.
+  for (const dir of favourHomeDrivePathList) {
+    try {
+      fs.accessSync(path.join(dir, '.kube', 'config'));
+
+      return dir;
+      // tslint:disable-next-line:no-empty
+    } catch (ignore) {
+    }
```

Comment:
```suggestion
    } catch {
      // Directory is not accessible.
    }
```
- TSLint is [deprecated](https://www.npmjs.com/package/tslint), and we're using ESLint with the appropriate plugins instead.
- Our [ESLint configuration](https://github.com/rancher-sandbox/rancher-desktop/blob/5ab09a3436be9a63df0610663982398bdbd57a88/.eslintrc.js#L26) has `no-empty` disabled completely (inherited from dashboard).
- for TSLint, `no-empty` can be configured to ignores blocks with comments; it's much better to add a comment on why it's fine instead.

Reply:
The tslint directives come directly from upstream, along with all the other code. Yes, we should fix this, even knowing that eventually all this code will be replaced with a call to the upstream function.

Reply:
Ah, right, there's an upstream. That makes it okay, I think, if it's a verbatim copy that we want to get rid of later.